docs-site/docs/development/development_processes/pull_request_workflow.md

```diff
@@ -0,0 +1,84 @@
+   * If a failing CI build seems to be unrelated to your contribution, you can try restarting the failing CI job, rebasing on top of target branch to bring in updates that may resolve the failure, or if it has not been fixed yet, ask a developer to help you fix the test.
+1. The PR contains a few logically organized commits, or has squashing commits enabled.
+1. The changes can merge without problems. If not, you should rebase if you're the only one working on your feature branch, otherwise merge the default branch into the PR branch.
+1. Only one specific issue is fixed or one specific feature is implemented. Do not combine things; send separate pull requests for each issue or feature.
+1. Migrations should do only one thing (for example, create a table, move data to a new table, or remove an old table) to aid retrying on failure.
+1. Contains functionality that other users will benefit from.
+1. Doesn't add configuration options or settings options since they complicate making and testing future changes.
+1. Changes do not degrade performance:
+   * Avoid repeated polling of endpoints that require a significant amount of overhead.
+   * Check for N + 1 queries via the SQL log.
+   * Avoid repeated access of the file system.
+1. If the pull request adds any new libraries (like gems or JavaScript libraries), they should conform to our Licensing guidelines. Also, make the reviewer aware of the new library and explain why you need it.
+
+## Definition of done
+
+If you contribute to IRIDA Next, please know that changes involve more than just code. We use the following [definition of done](https://www.agilealliance.org/glossary/definition-of-done).
+
+If a regression occurs, we prefer you revert the change. Your contribution is *incomplete* until you have made sure that it meets all these requirements.
+
+### Functionality
+
+1. Working and clean code that is commented where needed.
+1. Documented in the `/docs` directory.
+1. If your pull request adds one or more migrations, make sure to execute all migrations on a fresh database before the PR is reviewed. If the review leads to large changes in the PR, execute the migrations again after the review is complete.
+1. If your pull request adds new validations to existing models, to make sure the data processing is backwards compatible:
+   * Ask a IRIDA team member for assistance to execute the database query that checks the existing rows to ensure existing rows aren't impacted by the change.
+
+### Testing
+
+1. Unit, integration, and system tests that all pass on the CI server.
+1. Regressions and bugs are covered with tests that reduce the risk of the issue happening again.
+1. If your merge request adds one or more migrations, write tests for more complex migrations.
+
+### UI changes
+
+1. Use available components from the IRIDA Next Design System, Viral.
```

Comment:
The only thing I can think of is if a component does not exist, create a new PR with that component first.